# HG changeset patch
# User Rik <rik@octave.org>
# Date 1402434189 25200
#      Tue Jun 10 14:03:09 2014 -0700
# Node ID 87c3848cf3c0a5a16c0613ce8ef876d6900c6d54
# Parent  096b1a159d1fddf227b466c8b74fbfc05ab0a385
Fix bug when hggroup used with primitive graphic object (bug #42532).

* image.m, text.m, line.m, patch.m: __plt_get_axis_arg__ will return axis and
hggroup when 'parent' property is used.  Select the first returned object
which is the axes, rather than passing both axis and hggroup to further plot
subroutines.

diff --git a/scripts/image/image.m b/scripts/image/image.m
--- a/scripts/image/image.m
+++ b/scripts/image/image.m
@@ -106,16 +106,18 @@ function h = image (varargin)
   if (! isempty (hax))
     oldfig = get (0, "currentfigure");
   endif
   unwind_protect
     if (do_new)
       hax = newplot (hax);
     elseif (isempty (hax))
       hax = gca ();
+    else
+      hax = hax(1);
     endif
 
     htmp = __img__ (hax, do_new, x, y, img, varargin{chararg:end});
 
   unwind_protect_cleanup
     if (! isempty (oldfig))
       set (0, "currentfigure", oldfig);
     endif
diff --git a/scripts/plot/appearance/text.m b/scripts/plot/appearance/text.m
--- a/scripts/plot/appearance/text.m
+++ b/scripts/plot/appearance/text.m
@@ -124,16 +124,18 @@ function h = text (varargin)
     print_usage ();
   endif
 
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("text", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
+  else
+    hax = hax(1);
   endif
 
   ## Position argument may alse be in PROP/VAL pair
   idx = find (strcmpi (varargin, "position"), 1);
   if (idx)
     pos = varargin{idx+1};
     varargin(idx:idx+1) = [];
   else
diff --git a/scripts/plot/draw/line.m b/scripts/plot/draw/line.m
--- a/scripts/plot/draw/line.m
+++ b/scripts/plot/draw/line.m
@@ -44,16 +44,18 @@
 
 function h = line (varargin)
 
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("line", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
+  else
+    hax = hax(1);
   endif
 
   htmp = __line__ (hax, varargin{:});
 
   if (nargout > 0)
     h = htmp;
   endif
 
diff --git a/scripts/plot/draw/patch.m b/scripts/plot/draw/patch.m
--- a/scripts/plot/draw/patch.m
+++ b/scripts/plot/draw/patch.m
@@ -75,16 +75,18 @@
 ## Author: jwe
 
 function h = patch (varargin)
 
   [hax, varargin] = __plt_get_axis_arg__ ("patch", varargin{:});
   
   if (isempty (hax))
     hax = gca ();
+  else
+    hax = hax(1);
   endif
   
   [htmp, failed] = __patch__ (hax, varargin{:});
 
   if (failed)
     print_usage ();
   endif
 
